### PR TITLE
feat: add pagination params to storage service

### PR DESCRIPTION
**Pull request by mrlunin**
This is mainly an implementation for #1007.

Aries Askar supports pagination using offset and limit, so I added additional parameters `$offset` and `$limit` (with `$` to not overlap with some existing record tags) to the Storage service and added implementation to AskarStorageService.

I added this because we have server deployment and to avoid querying all records at once.

### CODE REVIEW

**Review by TimoGlastra — CHANGES_REQUESTED** [review]
I like the simplicity of the solution, but i don't think it should be added as tags. But rather as separate options

File: packages/core/src/storage/StorageService.ts [hunk 1]
```diff
@@ -12,7 +12,13 @@ interface AdvancedQuery<T extends BaseRecord> {
   $not?: Query<T>
 }
 
-export type Query<T extends BaseRecord<any, any, any>> = AdvancedQuery<T> | SimpleQuery<T>
+interface PaginationQuery {
```
Comment (genaris) [inline packages/core/src/storage/StorageService.ts:15]: I agree with @TimoGlastra in that maybe using an extra query options parameters would be cleaner. What about something like the following?

```ts
export type QueryOptions = {
  limit?: number
  offset?: number
}

findByQuery(agentContext: AgentContext, recordClass: BaseRecordConstructor<T>, query: Query<T>, queryOptions?: QueryOptions): Promise<T[]>
```

**Comment by mrlunin** [comment]
@TimoGlastra @genaris I've added an additional parameter `QueryOptions` to the `findByQuery`. Can you check the code?

**Review by TimoGlastra — APPROVED** [review]
I like it! Thanks @mrlunin 🙏

**Comment by TimoGlastra** [comment]
There's several tests failing @mrlunin, if you address these we can merge this :)

**Pull request merged**